robot/framesystem/framesystem_test.go

```diff
@@ -376,4 +373,68 @@ func TestServiceWithRemote(t *testing.T) {
 	test.That(t, err, test.ShouldBeNil)
 	t.Logf("frame system:\n%v", allParts)
 	test.That(t, r2.Close(context.Background()), test.ShouldBeNil)
+
+	// close remote
+	test.That(t, remoteRobot.Close(context.Background()), test.ShouldBeNil)
+	fsCfg, err = r2.FrameSystemConfig(context.Background())
+	test.That(t, err, test.ShouldBeNil)
+
+	// expected error as remote component is missing
+	_, err = referenceframe.NewFrameSystem("test", fsCfg.Parts, transforms)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "references non-existent parent")
+
+	// but without transforms, expect no errors
+	transforms = []*referenceframe.LinkInFrame{
+		referenceframe.NewLinkInFrame(referenceframe.World, testPose, "frame3", nil),
+	}
+	fs, err = referenceframe.NewFrameSystem("test", fsCfg.Parts, transforms)
+	test.That(t, err, test.ShouldBeNil)
+	t.Logf("frame system:\n%v", fsCfg)
+	test.That(t, fs.FrameNames(), test.ShouldHaveLength, 2)
```

Comment:
@raybjork I'm unable to get this test to work properly, and not super sure if it's something really dumb. FrameNames() end up with a length of 2, but it feels like it should still contain the world frame

Reply:
nvm I should've moved the r2 Close to later